clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1111,6 +1111,29 @@ LogicalResult cir::VecShuffleDynamicOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// VecExtractOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult cir::VecExtractOp::fold(FoldAdaptor adaptor) {
+  const auto vectorAttr =
+      llvm::dyn_cast_if_present<cir::ConstVectorAttr>(adaptor.getVec());
```

Comment:
isn't this always present? pure dyn_cast should be enough? 
```suggestion
  const auto vectorAttr =
      mlir::dyn_cast<cir::ConstVectorAttr>(adaptor.getVec());
```

The same holds for index below?

Reply:
I think you are right, i will create a small PR too to improve it in upstream

Reply:
Seem, not always present :D

https://github.com/llvm/clangir/actions/runs/15029476427

Reply:
Oh okay, sorry for that then :D

Reply:
It was nice to try :D